Terminate Warning with GCC 6.1

I'm seeing the following with the current master and GCC 6.1. Sorry I don't have time to look into it, but I at least wanted to log it.

<pre>
In file included from ./include/libmesh/libmesh_common.h:59:0,
                 from ./include/libmesh/libmesh.h:25,
                 from ../../pbauman/src/base/libmesh.C:20:
../../pbauman/src/base/libmesh.C: In destructor ‘virtual libMesh::LibMeshInit::~LibMeshInit()’:
./include/libmesh/libmesh_common.h:366:39: warning: throw will always call terminate() [-Wterminate]
     LIBMESH_THROW(libMesh::LogicError());                               \
                                       ^
./include/libmesh/libmesh_exceptions.h:150:37: note: in definition of macro ‘LIBMESH_THROW’
 #define LIBMESH_THROW(e) do { throw e; } while (0)
                                     ^
./include/libmesh/libmesh_common.h:288:7: note: in expansion of macro ‘libmesh_error_msg’
       libmesh_error_msg(msg);                                           \
       ^~~~~~~~~~~~~~~~~
./include/libmesh/libmesh_common.h:342:34: note: in expansion of macro ‘libmesh_assert_msg’
 #define libmesh_assert(asserted) libmesh_assert_msg(asserted, "")
                                  ^~~~~~~~~~~~~~~~~~
./include/libmesh/parallel.h:58:5: note: in expansion of macro ‘libmesh_assert’
     libmesh_assert((comm_obj).verify(std::string(__FILE__).size()));    \
     ^~~~~~~~~~~~~~
../../pbauman/src/base/libmesh.C:642:3: note: in expansion of macro ‘libmesh_parallel_only’
   libmesh_parallel_only(this->comm());
   ^~~~~~~~~~~~~~~~~~~~~
./include/libmesh/libmesh_common.h:366:39: note: in C++11 destructors default to noexcept
     LIBMESH_THROW(libMesh::LogicError());                               \
                                       ^
./include/libmesh/libmesh_exceptions.h:150:37: note: in definition of macro ‘LIBMESH_THROW’
 #define LIBMESH_THROW(e) do { throw e; } while (0)
                                     ^
./include/libmesh/libmesh_common.h:288:7: note: in expansion of macro ‘libmesh_error_msg’
       libmesh_error_msg(msg);                                           \
       ^~~~~~~~~~~~~~~~~
./include/libmesh/libmesh_common.h:342:34: note: in expansion of macro ‘libmesh_assert_msg’
 #define libmesh_assert(asserted) libmesh_assert_msg(asserted, "")
                                  ^~~~~~~~~~~~~~~~~~
./include/libmesh/parallel.h:58:5: note: in expansion of macro ‘libmesh_assert’
     libmesh_assert((comm_obj).verify(std::string(__FILE__).size()));    \
     ^~~~~~~~~~~~~~
../../pbauman/src/base/libmesh.C:642:3: note: in expansion of macro ‘libmesh_parallel_only’
   libmesh_parallel_only(this->comm());
</pre>

--- a/src/base/libmesh.C
+++ b/src/base/libmesh.C
@@ -638,10 +638,6 @@ LibMeshInit::~LibMeshInit()
   // Clear the thread task manager we started
   task_scheduler.reset();
 
-  // Let's be sure we properly close on every processor at once:
-  libmesh_parallel_only(this->comm());
-
-
   // Force the \p ReferenceCounter to print
   // its reference count information.  This allows
   // us to find memory leaks.  By default the

